test/CloneFactory.test.js

```diff
@@ -0,0 +1,65 @@
+const ERC20Mock = artifacts.require('ERC20Mock');
+const ERC20CloneFactory = artifacts.require('ERC20CloneFactory');
+
+const BigNumber = web3.BigNumber;
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('CloneFactory', function ([_, owner]) {
+  let master;
+  let clonedOne;
+  let clonedTwo;
+  let factory;
+
+  const _masterSupply = 64;
+  const _clonedSupplyOne = 128;
+  const _clonedSupplyTwo = 512;
+  const _updateValue = 32;
+
+  const initFactory = async function (factoryContract) {
+    const _factory = await factoryContract.new(master.address);
+    factory = {
+      createNew: async function (amount) {
+        const tx = await _factory.createNew(amount);
+        return ERC20Mock.at(tx.logs[0].args.cloneAddress);
+      }
+    };
+  };
+
+  describe('factory initialized with master contract', function () {
+    before(async function () {
+      master = await ERC20Mock.new(owner, _masterSupply);
+      await initFactory(ERC20CloneFactory);
+    });
+
+    describe('cloned instances created', function () {
```

Comment:
I think it'd probably make sense to check here that the proxy bytecode matches the one listed in the EIP, since that's the actual spec.